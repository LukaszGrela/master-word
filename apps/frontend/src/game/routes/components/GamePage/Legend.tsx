import { FC } from 'react';

export const Legend: FC<{
  incorrect: string;
  correct: string;
  misplaced: string;
}> = ({ correct, incorrect, misplaced }) => {
  return (
    <div className="legend">
      <span className="incorrect">{incorrect}</span>
      <span className="misplaced">{misplaced}</span>
      <span className="correct">{correct}</span>
    </div>
  );
};
