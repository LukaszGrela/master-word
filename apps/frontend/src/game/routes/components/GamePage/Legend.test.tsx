import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Legend } from './Legend';

describe('game', () => {
  describe('routes', () => {
    describe('components', () => {
      describe('GamePage', () => {
        describe('Legend', () => {
          it('renders', () => {
            const { container } = render(<Legend 
            correct='Correct'
            incorrect='Incorrect'
            misplaced='Misplaced' />)
            expect(container).toMatchSnapshot()
          });

        });
      });
    });
  });
});
