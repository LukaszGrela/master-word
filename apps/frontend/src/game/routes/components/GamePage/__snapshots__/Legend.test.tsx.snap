// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > routes > components > GamePage > Legend > renders 1`] = `
<div>
  <div
    class="legend"
  >
    <span
      class="incorrect"
    >
      Incorrect
    </span>
    <span
      class="misplaced"
    >
      Misplaced
    </span>
    <span
      class="correct"
    >
      Correct
    </span>
  </div>
</div>
`;
